src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SemanticFacts/CSharpSemanticFacts.cs

```diff
@@ -60,10 +60,21 @@ public ISymbol GetDeclaredSymbol(SemanticModel semanticModel, SyntaxToken token,
                 var symbol = semanticModel.GetDeclaredSymbol(ancestor, cancellationToken);
                 if (symbol != null)
                 {
-                    // The token may be part of a larger name (for example, `int` in `public static operator int[](Goo g);`.
-                    // So check if the symbol's location encompasses the span of the token we're asking about.
-                    if (symbol.Locations.Any(loc => loc.SourceTree == location.SourceTree && loc.SourceSpan.Contains(location.SourceSpan)))
-                        return symbol;
+                    if (symbol is IMethodSymbol { MethodKind: MethodKind.UserDefinedOperator })
+                    {
+                        // The token may be part of a larger name (for example, `int` in `public static operator int[](Goo g);`.
+                        // So check if the symbol's location encompasses the span of the token we're asking about.
+                        if (symbol.Locations.Any(loc => loc.SourceTree == location.SourceTree && loc.SourceSpan.Contains(location.SourceSpan)))
+                            return symbol;
+                    }
+                    else
+                    {
+                        // For any other symbols, we only care if the name directly matches the span of the token
+                        if (symbol.Locations.Contains(location))
+                        {
+                            return symbol;
+                        }
```

Comment:
```suggestion
                            return symbol;
```

to match the formatting you have above.

Reply:
Fixed.